test/unit/overlay.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+var fixtures = require('../fixtures');
+var fs = require('fs');
+var sharp = require('../../index');
+
+sharp.cache(0);
+
+// Main
+describe('Overlays', function() {
+  it('Overlay transparent PNG on solid background', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
+
+        fixtures.assertSimilar(fixtures.expected('alpha-layer-01.png'), data, done);
+      });
+  });
+
+  it('Composite three transparent PNGs into one', function(done) {
+    sharp(fixtures.inputPngOverlayLayer0)
+      .overlayWith(fixtures.inputPngOverlayLayer1)
+      .toBuffer(function (error, data, info) {
+        if (error) return done(error);
+
+        sharp(data)
+          .overlayWith(fixtures.inputPngOverlayLayer2)
+          .toBuffer(function (error, data, info) {
+            if (error) return done(error);
+
+            // TODO: Remove once this fails
+            fixtures.assertSimilar(fixtures.expected('alpha-layer-012-low-alpha.png'), data, function (error, result) {
```

Comment:
The image fingerprinting method normalises image luminance so won't be able to detect all differences. It's worth trying your suggestion of exposing assertSimilar's threshold.